octoprint_mrbeam/iobeam/airfilter.py

```diff
@@ -260,6 +288,27 @@ def set_pressure(
         elif self._pressure1 is not None and self.model_id in self.AIRFILTER2_MODELS:
             self._last_pressure_values.append(self._pressure1)
 
+    def set_device(self, device):
+        """
+
+        Args:
+            device (exhaust.Device):
+
+        Returns:
+
+        """
+        if (
+            self._external_power != device.ext_power
+            and device.ext_power
+            and device.serial_num == 0
+        ):
+            self._logger.info(
+                "Exhaust fan is now connected to external power -> reset exhaust"
+            )
+            self._iobeam.reset_exhaust()
+            return None
```

Comment:
I would add a return at the end of the method too, or just have the last line in an `else`, I think it's more clear